pkg/cloudprovider/providers/aws/volumes.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"net/url"
+	"strings"
+)
+
+// awsVolumeID represents the ID of the volume in the AWS API, e.g. vol-12345678a
+type awsVolumeID string
+
+func (i awsVolumeID) awsString() *string {
+	return aws.String(string(i))
+}
+
+// kubernetesVolumeSpec represents the spec for a volume in the kubernetes API;
+// a few forms are recognized:
+//  * aws://<zone>/<id>
+//  * aws:///<id>
+//  * <id>
+type KubernetesVolumeID string
+
+func mapVolumeSpecToAwsID(name KubernetesVolumeID) (awsVolumeID, error) {
```

Comment:
It's nothing serious, however naming style in whole AWS provider is mixed: sometimes is `AWS` (`newAWSDisk`), sometimes `Aws` (`mapVolumeSpecToAwsID`), ID/Id should probably follow the same pattern.